test/server.test.js

```diff
@@ -1,150 +1,144 @@
+  )
+})
 
+test('listen should accept undefined port', async t => {
   const fastify = Fastify()
-  t.teardown(fastify.close.bind(fastify))
-  fastify.listen({ port: undefined }, (err) => {
-    t.error(err)
-  })
-})
+  t.after(() => fastify.close())
 
-test('listen should accept stringified number port', t => {
-  t.plan(1)
+  await t.assert.doesNotReject(
+    fastify.listen({ port: undefined })
+  )
+})
 
+test('listen should accept stringified number port', async t => {
   const fastify = Fastify()
-  t.teardown(fastify.close.bind(fastify))
-  fastify.listen({ port: '1234' }, (err) => {
-    t.error(err)
-  })
-})
+  t.after(() => fastify.close())
 
-test('listen should accept log text resolution function', t => {
-  t.plan(3)
+  await t.assert.doesNotReject(
+    fastify.listen({ port: '1234' })
+  )
+})
 
+test('listen should accept log text resolution function', async t => {
   const fastify = Fastify()
-  t.teardown(fastify.close.bind(fastify))
-  fastify.listen({
-    host: '127.0.0.1',
-    port: '1234',
-    listenTextResolver: (address) => {
-      t.equal(address, 'http://127.0.0.1:1234')
-      t.pass('executed')
-      return 'hardcoded text'
-    }
-  }, (err) => {
-    t.error(err)
-  })
+  t.after(() => fastify.close())
+
+  await t.assert.doesNotReject(
+    fastify.listen({
+      host: '127.0.0.1',
+      port: '1234',
+      listenTextResolver: (address) => {
+        t.assert.strictEqual(address, 'http://127.0.0.1:1234')
+        return 'hardcoded text'
+      }
+    })
+  )
 })
 
 test('listen should reject string port', async (t) => {
-  t.plan(2)
   const fastify = Fastify()
-  t.teardown(fastify.close.bind(fastify))
+  t.after(() => fastify.close())
 
   try {
     await fastify.listen({ port: 'hello-world' })
   } catch (error) {
-    t.equal(error.code, 'ERR_SOCKET_BAD_PORT')
+    t.assert.strictEqual(error.code, 'ERR_SOCKET_BAD_PORT')
   }
 
   try {
     await fastify.listen({ port: '1234hello' })
   } catch (error) {
-    t.equal(error.code, 'ERR_SOCKET_BAD_PORT')
+    t.assert.strictEqual(error.code, 'ERR_SOCKET_BAD_PORT')
   }
 })
 
-test('Test for hostname and port', t => {
+test('Test for hostname and port', (t, end) => {
   const app = Fastify()
-  t.teardown(app.close.bind(app))
+  t.after(() => app.close())
   app.get('/host', (req, res) => {
     const host = 'localhost:8000'
-    t.equal(req.host, host)
-    t.equal(req.hostname, req.host.split(':')[0])
-    t.equal(req.port, Number(req.host.split(':')[1]))
+    t.assert.strictEqual(req.host, host)
+    t.assert.strictEqual(req.hostname, req.host.split(':')[0])
+    t.assert.strictEqual(req.port, Number(req.host.split(':')[1]))
     res.send('ok')
   })
+
   app.listen({ port: 8000 }, () => {
-    sget('http://localhost:8000/host', () => { t.end() })
+    sget('http://localhost:8000/host', () => { end() })
   })
 })
 
-test('abort signal', t => {
-  t.test('listen should not start server', t => {
+describe('abort signal', () => {
```

Comment:
We don't need to use `describe`. 
```suggestion
test('abort signal', async () => {
```

Also, each subtest should be like:

```js
await test('listen should not start server', (t, end) => { end() })
```

Reply:
Hey @jsumners, no problem at all, I'm pushing the changes 😁   
Could I ask about the rationale behind this choice? 😁

I ask because, IMHO, the describe / test / it approach is cleaner than the `await t.test`.

Reply:
We don't use it anywhere else. And I find it noisy.

Reply:
It makes perfect sense, I was just curious 🚀 
Thanks for your review!

Reply:
pushed the changes in the meantime!